### PR TITLE
MAINT: Make f2py generated file not contain the (local) date.

### PR DESCRIPTION
**Pull request by bmwiedemann**
While working on the reproducible builds effort, I found that when building the python-scipy package for openSUSE Linux, there were slight differences between each build that can be avoided with this patch to 
Allow to override build date
and use gmtime in order to be independent of timezone
in order to make builds reproducible.
See https://reproducible-builds.org/ for why this is good
and https://reproducible-builds.org/specs/source-date-epoch/
for the definition of this variable.

### CODE REVIEW

**Comment by bashtage** [comment]
You need to change your PR title.

**Comment by charris** [comment]
The lines with "Revision" and "Date" were for subversion. I'd just remove all of them, including also the time and the outdated link.

**Comment by charris** [comment]
And you need to completely redo your commit message. See `doc/source/dev/gitwash/development_workflow.rst`.

File: numpy/f2py/rules.py [hunk 1]
```diff
@@ -114,7 +114,7 @@
  * f2py is a Fortran to Python Interface Generator (FPIG), Second Edition,
  * written by Pearu Peterson <pearu@cens.ioc.ee>.
  * See http://cens.ioc.ee/projects/f2py2e/
- * Generation date: """ + time.asctime(time.localtime(time.time())) + """
+ * Generation date: """ + time.asctime(time.gmtime(int(os.environ.get('SOURCE_DATE_EPOCH', time.time())))) + """
  * $R""" + """evision:$
```
Comment (charris) [inline numpy/f2py/rules.py]: Could you remove this line and the one below?

File: numpy/f2py/rules.py [hunk 2]
```diff
@@ -114,7 +114,7 @@
  * f2py is a Fortran to Python Interface Generator (FPIG), Second Edition,
  * written by Pearu Peterson <pearu@cens.ioc.ee>.
  * See http://cens.ioc.ee/projects/f2py2e/
```
Comment (charris) [inline numpy/f2py/rules.py]: Remove this line also.

File: numpy/f2py/rules.py [hunk 3]
```diff
@@ -114,7 +114,7 @@
...
- * Generation date: """ + time.asctime(time.localtime(time.time())) + """
+ * Generation date: """ + time.asctime(time.gmtime(int(os.environ.get('SOURCE_DATE_EPOCH', time.time())))) + """
```
Comment (charris) [inline numpy/f2py/rules.py]: If you are going to keep this, could you break it so that the line length is < 80 chars?

**Comment by charris** [comment]
I note that f2py is now released together with NumPy, so its own version number no longer tracks changes. If reproducibility requires a known version something more might be required.

**Comment by charris** [comment]
Merged, thanks @bmwiedemann.

There may be more work needed here, but I don't know the precise aims that you have in the reproducibile build project.

**Comment by bmwiedemann** [comment]
For now, my goal is for all 11800 openSUSE source packages to be able to build twice and produce identical binary packages (which was not the case for python-scipy before this patch, because of the timestamps).
Version numbers usually do not matter there, because they do not vary between builds.

**Pull request merged**